contracts/PrivatePool.sol

```diff
@@ -0,0 +1,995 @@
+                    datas[i],
+                    (IERC20, address, int256)
+                );
+                bentoBox.transfer(
+                    token,
+                    msg.sender,
+                    to,
+                    _num(share, value1, value2)
+                );
+            } else if (action == ACTION_BENTO_TRANSFER_MULTIPLE) {
+                (
+                    IERC20 token,
+                    address[] memory tos,
+                    uint256[] memory shares
+                ) = abi.decode(datas[i], (IERC20, address[], uint256[]));
+                bentoBox.transferMultiple(token, msg.sender, tos, shares);
+            } else if (action == ACTION_CALL) {
+                (bytes memory returnData, uint8 returnValues) = _call(
+                    values[i],
+                    datas[i],
+                    value1,
+                    value2
+                );
+
+                if (returnValues == 1) {
+                    (value1) = abi.decode(returnData, (uint256));
+                } else if (returnValues == 2) {
+                    (value1, value2) = abi.decode(
+                        returnData,
+                        (uint256, uint256)
+                    );
+                }
+            } else if (action == ACTION_GET_REPAY_SHARE) {
+                int256 part = abi.decode(datas[i], (int256));
+                value1 = bentoBox.toShare(
+                    asset,
+                    totalDebt.toElastic(_num(part, value1, value2), true),
+                    true
+                );
+            } else if (action == ACTION_GET_REPAY_PART) {
+                int256 amount = abi.decode(datas[i], (int256));
+                value1 = totalDebt.toBase(_num(amount, value1, value2), false);
+            }
+        }
+
+        if (status.needsSolvencyCheck) {
+            require(
+                _isSolvent(msg.sender, exchangeRate),
+                "PrivatePool: borrower insolvent"
+            );
+        }
+    }
+
+    /// @notice Handles the liquidation of borrowers' balances, once the borrowers' amount of collateral is too low.
+    /// @param borrowers An array of borrower addresses.
+    /// @param maxDebtParts A one-to-one mapping to `borrowers`, contains maximum part (not token amount) of the debt that will be liquidated of the respective borrower.
+    /// @param to Address of the receiver if `swapper` is zero.
+    /// @param swapper Contract address of the `ISimpleSwapper` implementation.
+    function liquidate(
+        address[] calldata borrowers,
+        uint256[] calldata maxDebtParts,
+        address to,
+        ISimpleSwapper swapper
+    ) public {
+        // Oracle can fail but we still need to allow liquidations
+        (, uint256 _exchangeRate) = updateExchangeRate();
+        accrue();
+
+        AccrueInfo memory _accrueInfo = accrueInfo;
+        require(
+            block.timestamp >= _accrueInfo.NO_LIQUIDATIONS_BEFORE,
```

Comment:
I think if we still have an oracle we still need a liquidation if it goes insolvent, it should just also liquidate if the time period is up

Reply:
after the time period is up, liquidations should be possible even if the position is still solvent.